### ISSUE TITLE
Upload long text instead of posting it to channel

### ISSUE THREAD

**Issue by moviuro** [weechat/weechat#545]
Hi all!

Instead of
```
[Paste 44195820 lines ? [ctrl-Y] Yes [ctrl-N] No]
```
I think it would be best to also ask if I want to upload it to a cool paste site (http://sprunge.us or http://ix.io) and paste that link in the channel. And also, make it default behavior?
```
[Paste 44195820 lines ? [ctrl-Y] Yes, upload to a pastebin [ctrl-F] Flood the channel [ctrl-N] No]
```

Best regards,
Note: the aforementioned pastebins work from cli:
```sh
< /path/to/file | curl -F 'sprunge=<-' http://sprunge.us
< /path/to/file | curl -F 'f:1=<-' ix.io
```

**Comment by HS-157** [comment]
Hello,

We could also envision asking which one of many sites should be used at each specific occurence of multi-line pasting (e.g. sprunge, ix, zerobin, ...)

**Comment by Mikaela** [comment]
I think that should be done by script instead of WeeChat core.

The pastebins can have many issues like be down or have geo-restrictions (e.g. sprungeus was returning some Google Apps messages about it being not allowed in Germany to my DigitalOcean droplet, at the moment it seems to work).

**Comment by moviuro** [comment]
@Mikaela I have no idea of how that would work, though. Could you link to the wiki/knowledge base?

**Comment by sim642** [comment]
Of course it makes sense for scripts to handle this as that leaves infinitely more options but currently I think it's not possible. There is some hook_signal for input_pending_paste but that tells you nothing more than long paste started or finished, which isn't really helpful in implementing such paste site functionality in a script. The way I see it, there'd have to be more hooks to allow handling different cases and actual possibility to access the copy buffer.

**Comment by flashcode** [comment]
Yes, I'll add a hook (probably hsignal) so that plugins/scripts can catch the pasted text, do what they want with it, and then discard it (so that WeeChat will not ask anything to the user).

**Comment by sim642** [comment]
I was thinking it could be built into the current user asking mechanism quite nicely. That "ctrl-Y, ctrl-N" line could still pop up and after you choose either the hsignal will be fired with pasted text and a boolean telling whether you accepted or not.

**Comment by Mikaela** [comment]
>  I have no idea of how that would work, though. Could you link to the wiki/knowledge base?

I am not fully sure what you mean, but if I understood correctly, the scripting guide is at https://weechat.org/files/doc/stable/weechat_scripting.en.html

**Comment by batalooppak** [comment]
If you like I made a script for what you need. You can find it in here.
https://github.com/batalooppak/pastero
